### PR TITLE
add a msys2-runtime-3.3 package

### PR DESCRIPTION
**Pull request by rimrul**
The first step of phase 2 of the current timeline(https://github.com/git-for-windows/git/issues/4279#issue-1577622335)

### CODE REVIEW

File: msys2-runtime-3.3/PKGBUILD [hunk 1]
```diff
@@ -1,10 +1,10 @@
...
 
-pkgbase=msys2-runtime
-pkgname=('msys2-runtime' 'msys2-runtime-devel')
+pkgbase=msys2-runtime-3.3
+pkgname=('msys2-runtime-3.3' 'msys2-runtime-3.3-devel')
 pkgver=3.3.6
-pkgrel=4
+pkgrel=1
 pkgdesc="Cygwin POSIX emulation engine"
 arch=('i686' 'x86_64')
```
Comment (rimrul) [inline msys2-runtime-3.3/PKGBUILD:9]: It might make sense to mark this as `i686` only.

**Review by dscho — COMMENTED** [review]
These changes look pretty good so far, thank you _so much_ for working on this @rimrul!

I allowed myself to add a commit to trigger i686 builds for PRs containing the needle `msys2-runtime-3.3` in their title, as this here PR does. I hope to get an artifact that I can then test.

File: msys2-runtime-3.3/update-patches.sh [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+git rm 0*.patch ||
+die "Could not remove previous patches"
+
+base_tag=refs/tags/cygwin-"$(sed -ne 'y/./_/' -e 's/^pkgver=//p' <PKGBUILD)"-release
+source_url=$(sed -ne 's/^source=\([^:]\+::\)\?["'\'']\?\([^"'\''#?=&,;[:space:]]\+[^)"'\''#?=&,;[:space:]]\).*/\2/p' <PKGBUILD)
+
+git -C src/msys2-runtime fetch --no-tags "$source_url" "$base_tag:$base_tag"
+
+merging_rebase_start="$(git -C src/msys2-runtime \
+    rev-parse --verify --quiet HEAD^{/Start.the.merging.rebase})"
```
[the remark this answers is not in the document]
Comment (rimrul) [reply msys2-runtime-3.3/update-patches.sh:26]: > Therefore, I propose to branch off, say, `msys2-gfw-3.3-branch` from the current tip in `git-for-windows/msys2-runtime` and add an `update-patches.sh` to the `msys2-runtime-3.3/` directory that we can use to update the (i686-only) MSYS2 runtime v3.3.* in the future.

Makes sense.

File: msys2-runtime-3.3/tests/do_new.bat [hunk 3]
```diff
@@ -0,0 +1,6 @@
+set MSYS64=%1
```
[the remark this answers is not in the document]
Comment (rimrul) [reply msys2-runtime-3.3/tests/do_new.bat:1]: I'm happy to get rid of them. I just copied them over from `msys2-runtime`

[the remark this answers is not in the document]
Comment (dscho) [reply msys2-runtime-3.3/tests/do_new.bat:1]: Oh wow! I had not realized that they're also in `msys2-runtime/`... :-( Maybe get rid of that subdirectory there first?

File: msys2-runtime-3.3/update-patches.sh (hunk 2 above)
[the remark this answers is not in the document]
Comment (dscho) [reply msys2-runtime-3.3/update-patches.sh]: > I propose to branch off, say, `msys2-gfw-3.3-branch` from the current tip

[Done](https://github.com/git-for-windows/msys2-runtime/commits/msys2-gfw-3.3-branch).

I also updated the `update-patches.sh` script, could you please double-check my work, @rimrul?

File: msys2-runtime-3.3/tests/do_new.bat (hunk 3 above)
[the remark this answers is not in the document]
Comment (dscho) [reply msys2-runtime-3.3/tests/do_new.bat]: > Maybe get rid of that subdirectory there first?

I did that: 3c4d00dd93f48375247b1f8957ae68488017afad (plan is to move this to the beginning of the topic branch using an interactive rebase before calling this here PR ready-to-merge).

**Comment by dscho** [comment]
@rimrul from my side, I consider this one done (modulo the final interactive rebase). Anything else you want to see before flipping this to "ready-for review"?

File: msys2-runtime-3.3/update-patches.sh (hunk 2 above)
[the remark this answers is not in the document]
Comment (rimrul) [reply msys2-runtime-3.3/update-patches.sh]: Looks good to me.

**Review by dscho — APPROVED** [review]
Yeah!

**Comment by dscho** [comment]
I guess _technically_ we do not need the x86_64 builds, but what's the harm, eh? 😁

**Comment by rimrul** [comment]
Yes, that should be fine.

**Comment by dscho** [comment]
I _think_ the next step should be a PR in `git-sdk-32` that teaches the `sync` job (or more correctly: [`update-via-pacman.ps1`](https://github.com/git-for-windows/git-sdk-32/blob/main/update-via-pacman.ps1) to replace the `msys2-runtime` (if installed) package by the `msys2-runtime-3.3`.

**Comment by dscho** [comment]
> I guess _technically_ we do not need the x86_64 builds, but what's the harm, eh? 😁

So now I know what the harm is: https://github.com/git-for-windows/git-sdk-64/actions/runs/4921830149/jobs/8792046619 says this:

```
[...]
:: Starting core system upgrade...
warning: terminate other MSYS2 programs before proceeding
resolving dependencies...
looking for conflicting packages...
error: failed to prepare transaction (could not satisfy dependencies)
:: installing msys2-runtime (3.4.6-2) breaks dependency 'msys2-runtime=3.3.6' required by msys2-runtime-devel
[...]
```

Same problem [in the `sync` job of `git-sdk-arm64`](https://github.com/git-for-windows/git-sdk-arm64/actions/runs/4921935626/jobs/8792249286#step:3:53).

This suggests to me that the `-3.3-` in `msys2-runtime-3.3-3.3.6-1` is causing problems.

I fear that we'll have to [`repo-remove`](https://archlinux.org/pacman/repo-add.8.html) it ASAP, and hope that simply renaming the package to `msys2-runtime3.3` will work...

**Comment by dscho** [comment]
Hmm. MSYS2 does not seem to have any problems with this package name! See https://packages.msys2.org/package/msys2-runtime-3.3-devel?repo=msys&variant=x86_64

Also:

```console
$ tar tvf /var/lib/pacman/sync/msys.db | grep msys2-runtime
drwxr-xr-x repo/repo         0 2023-04-02 10:39 msys2-runtime-3.3-3.3.6-5/
-rw-r--r-- repo/repo      1514 2023-04-02 10:39 msys2-runtime-3.3-3.3.6-5/desc
drwxr-xr-x repo/repo         0 2023-04-02 10:39 msys2-runtime-3.3-devel-3.3.6-5/
-rw-r--r-- repo/repo      1562 2023-04-02 10:39 msys2-runtime-3.3-devel-3.3.6-5/desc
drwxr-xr-x repo/repo         0 2023-04-02 10:39 msys2-runtime-3.4.6-2/
-rw-r--r-- repo/repo      1491 2023-04-02 10:39 msys2-runtime-3.4.6-2/desc
drwxr-xr-x repo/repo         0 2023-04-02 10:39 msys2-runtime-devel-3.4.6-2/
-rw-r--r-- repo/repo      1534 2023-04-02 10:39 msys2-runtime-devel-3.4.6-2/desc
```

So the problem must lie with Git for Windows' tooling... I suspect [`pacman-helper.sh`](https://github.com/git-for-windows/build-extra/blob/HEAD/pacman-helper.sh)...

**Comment by dscho** [comment]
> So the problem must lie with Git for Windows' tooling... I suspect [`pacman-helper.sh`](https://github.com/git-for-windows/build-extra/blob/HEAD/pacman-helper.sh)...

Indeed, it looks that way. I consistently used the pattern `${dirname%%-[0-9]*}` (e.g. [here](https://github.com/git-for-windows/build-extra/blob/294695a6833908b9a3f088a35448caaad73e40f9/pacman-helper.sh#L434)) in Git for Windows' tooling to deduce the package name from the directory name in the tar files `/var/lib/pacman/sync/*.db`. But Pacman actually uses `${dirname%-*-*}`, see e.g. [here](https://github.com/git-for-windows/git-sdk-64/blob/779ad660d4d37517e7417b2d04d1172757586b94/usr/bin/repo-remove#L127).

Gotta fix that in All The Places. And then fix Git for Windows' Pacman databases...

**Comment by dscho** [comment]
> Gotta fix that in All The Places.

See https://github.com/git-for-windows/build-extra/pull/499 for that fix.

> And then fix Git for Windows' Pacman databases...

Done in https://github.com/git-for-windows/git-for-windows-automation/actions/runs/4930413416/jobs/8811298893. The `git-sdk-arm64` sync [worked](https://github.com/git-for-windows/git-sdk-arm64/actions/runs/4921935626/jobs/8811327952) and I started the [re-run of the `git-sdk-64` sync](https://github.com/git-for-windows/git-sdk-64/actions/runs/4921830149/jobs/8811891538), too.

**Pull request merged**